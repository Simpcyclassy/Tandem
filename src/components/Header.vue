<template>
  <v-row class="header-container">
    <div class="d-flex">
      <v-btn
        icon
        to="/"
        id="no-background-hover"
        @click="resetCounter"
      >
          <v-img
            alt="Tandem Logo"
            class="shrink mr-2"
            contain
            src="../assets/images/logo.png"
            transition="scale-transition"
            width="170"
          />
      </v-btn>
    </div>
  </v-row>
</template>

<script lang="ts">
import Vue from 'vue'
import { MutationTypes } from '../store/mutation-types'

export default Vue.extend({
  name: 'Header',

  data: () => ({
  }),
  methods: {
    resetCounter () {
      this.$store.commit(MutationTypes.RESET_STATE, 0)
    }
  }
})
</script>

<style scoped lang="scss">
.header-container {
  margin-top: 40px;
  margin-left: 150px;
}
#no-background-hover::before {
   background-color: transparent !important;
}
</style>
