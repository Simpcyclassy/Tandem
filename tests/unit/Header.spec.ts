import { createLocalVue, shallowMount } from '@vue/test-utils'
import Header from '@/components/Header.vue'
import Vuetify from 'vuetify'

describe('Header.vue', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()    
  })

  const wrapper = shallowMount(Header, {
    localVue,
    vuetify,
  })

  it(`is a vue instance and match snapshot`, () => {

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.exists()).toBeTruthy()
  })

  it(`should find v-btn`, () => {
    expect(wrapper.findComponent({ name: 'v-btn' }).exists()).toBe(true);
  })
})
