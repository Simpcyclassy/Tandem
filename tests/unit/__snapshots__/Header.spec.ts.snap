// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue is a vue instance and match snapshot 1`] = `
<v-row-stub tag="div" class="header-container">
  <div class="d-flex">
    <v-btn-stub tag="button" activeclass="" to="/" icon="true" type="button" id="no-background-hover">
      <v-img-stub width="170" alt="Tandem Logo" contain="true" options="[object Object]" position="center center" src="../assets/images/logo.png" transition="scale-transition" class="shrink mr-2"></v-img-stub>
    </v-btn-stub>
  </div>
</v-row-stub>
`;
